src/js/osweb/items/generic_response.js

```diff
@@ -126,16 +126,14 @@ export default class GenericResponse extends Item {
   prepare_duration_keypress () {
     // Prepare a keyboard duration.
     this._keyboard = new Keyboard(this.experiment)
-    var final_duration = (this._timeout !== -1) ? this._timeout : this._duration
-    this._keyboard._set_config(final_duration, this._allowed_responses)
+    this._final_duration = (this._timeout !== -1) ? this._timeout : this._duration
   }
 
   /** Prepare the system for a mouseclick duration interval. */
   prepare_duration_mouseclick () {
     // Prepare a mouseclick duration.
     this._mouse = new Mouse(this.experiment)
-    var final_duration = (this._timeout !== -1) ? this._timeout : this._duration
-    this._mouse._set_config(final_duration, this._allowed_responses, false)

```

Comment:
Are these and kb._set_config still necessary after this change? If not, I would clean them up/remove them too.

Reply:
Yes, they are just called later on.